src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -628,6 +628,42 @@ public bool PersistSecurityInfo
         }
         bool _persistSecurityInfo;
 
+        /// <summary>
+        /// Gets or sets a Boolean value that indicates if NpgsqlCommand parameters should be included (false) or excluded (true) within logging.
+        /// </summary>
+        [Category("Security")]
+        [Description("When enabled prevents logging of command parameter values.")]
+        [DisplayName("Exclude Parameters From Logs")]
+        [NpgsqlConnectionStringProperty]
+        public bool ExcludeParametersFromLogs
+        {
+            get => _excludeParametersFromLogs;
+            set
+            {
+                _excludeParametersFromLogs = value;
+                SetValue(nameof(ExcludeParametersFromLogs), value);
+            }
+        }
+        bool _excludeParametersFromLogs;
+
+        /// <summary>
+        /// Gets or sets a Boolean value that indicates if the Postgres DETAIL field is included in raised exceptions.
+        /// </summary>
+        /// [Category("Security")]
+        [Description("When enabled prevents excludes the Postgres DETAIL response from raised exceptions. This DETAIL message often contains user data that may be sensative.")]
+        [DisplayName("Suppress Postgres Error Detail")]
+        [NpgsqlConnectionStringProperty]
+        public bool SuppressDetailInPostgressError
```

Comment:
Probably it would be better to call it `SuppressDetailedExceptions` or something like this. It should be short enough but meaningful.